editor/src/clj/editor/handler.clj

```diff
@@ -403,12 +403,13 @@
   Returns either a non-empty vector or nil"
   [[{:keys [command]} :as handler+command-context]]
   (when-let [opts (options handler+command-context)]
-    (->> opts
-         (e/mapcat flatten-menu-item-tree)
-         (e/remove #(= :separator (:label %)))
-         (e/filter #(= command (:command %)))
-         vec
-         coll/not-empty)))
+    (when-let [flat-opts (->> opts
+                              (e/mapcat flatten-menu-item-tree)
+                              (e/remove #(= :separator (:label %)))
+                              (e/filter #(= command (:command %)))
+                              vec
+                              coll/not-empty)]
+      (with-meta flat-opts (meta opts)))))
```

Comment:
This preserves the designated sort function for options if it exists (e.g., when we show a dialog with a list of options for adding a component type to a game object or collision object)